src/TraceEvent/TraceEventSession.cs

```diff
@@ -2250,24 +2251,22 @@ private void EnsureStarted(TraceEventNativeMethods.EVENT_TRACE_PROPERTIES* prope
                 properties = GetProperties(propertiesBuff);
             }
 
-            int retCode = TraceEventNativeMethods.StartTraceW(out m_SessionHandle, m_SessionName, properties);
+            int retCode = TraceEventNativeMethods.StartTrace(out m_SessionHandle, m_SessionName, properties);
             if (retCode == 0xB7 && m_ResartIfExist)      // STIERR_HANDLEEXISTS
             {
                 m_restarted = true;
                 Stop();
                 m_Stopped = false;
-                Thread.Sleep(100);  // Give it some time to stop. 
-                retCode = TraceEventNativeMethods.StartTraceW(out m_SessionHandle, m_SessionName, properties);
+                Thread.Sleep(100);  // Give it some time to stop.
+                retCode = TraceEventNativeMethods.StartTrace(out m_SessionHandle, m_SessionName, properties);
             }
             if (retCode == 5 && OperatingSystemVersion.AtLeast(51))     // On Vista and we get a 'Accessed Denied' message
             {
-                m_SessionHandle = TraceEventNativeMethods.INVALID_HANDLE_VALUE; // StartTrace sets to 0 on failure.  We use INVALID_HANDLE_VALUE to represent failure
```

Comment:
If `m_sessionHandle` currently gets set to 0 on failure, and then we update it to `INVALID_HANDLE_VALUE`, the behavior here will change because `TraceEventNativeMethods.StartTrace` will determine that the return code is non-zero, and that the `sessionHandle` is non-zero, and will call `CloseTrace` on it.  We should avoid the call to `CloseTrace` in these failure cases.

Reply:
This is a case I considered and I believe the new `StartTrace` overload handles this case carefully and correctly.
It returns a new `SafeTraceHandle` only if retCode is zero (success).
Otherwise (the P/Invoke StartTraceW returned a non-zero value), we set the `out` param to null.
There's an edge case, also handled, where we have a non-zero retCode but a valid TRACEHANDLE; the StartTrace overload will call CloseTrace prior to returning (with a null out param).

Reply:
StartTraceW docs say "Do not use this handle if the function fails. Do not compare the session handle to INVALID_HANDLE_VALUE. The session handle is 0 if the handle is not valid."
I believe that means we shouldn't call CloseHandle if retCode is non-zero. I will fix that in the StartTrace overload.

Reply:
Fixed in latest iteration (removed the CloseHandle from the StartTrace overload)